### ISSUE TITLE
Investigate Electrino and Weex and Flutter and NodeGUI etc as Electron alternatives for Desktop

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#263]
### Problem

["Electron is Flash for the desktop"](https://josephg.com/blog/electron-is-flash-for-the-desktop/), although see this [sort-of rebuttal](https://github.com/workflowproducts/postage/blob/master/Facts_About_Electron_Performance.md).

### Solution

167 kB app vs 115 MB by using OS's native browser.

https://github.com/pojala/electrino (note, see this issue on merging Electrino with MacGap: `https://github.com/pojala/electrino/issues/10`)

Downside is fewer desktop integration features, and the fact that it uses Microsoft Edge on windows... But that might not turn out to be a big deal?

Another possibility is [Weex](https://blog.learnstartup.net/p/native-mobile-apps-weex-vuejs-2-0/), which is like React Native, except for VueJS.

Still yet another possibility is [volt](https://volt.ws), which account to its site will be open sourced in 2021.

EDIT (2026-06): [Deno Desktop](https://docs.deno.com/runtime/desktop/) ! ([hn](https://news.ycombinator.com/item?id=48626137)). Problem: doesn't support mobile.

EDIT: see also: https://jlongster.com/secret-of-good-electron-apps

EDIT: also see [NodeGUI](https://hibbard.eu/node-gui/) -> https://github.com/nodegui/nodegui -> https://github.com/nodegui/vue-nodegui

EDIT: [Neutralino!](https://neutralino.js.org/)

EDIT: [Muon](https://github.com/ImVexed/muon)

EDIT: [Tauri](https://github.com/tauri-apps/tauri) !! => https://tauri.studio/en/ - See: [Tauri vs Electron](https://www.levminer.com/blog/tauri-vs-electron), [example app](https://github.com/cardo-podcast/cardo)

- Thread on native push notifications: https://github.com/tauri-apps/tauri/issues/11651

EDIT: [Capacitor](https://capacitorjs.com/) by Ionic (used to be [Cordova](https://capacitorjs.com/cordova) I think?)

EDIT: [Sciter](https://sciter.com/) ! => https://sciter.com/

EDIT: **[Socket SDK](https://sockets.sh/) !!** - looks really promising!! Desktop & mobile & P2P!

EDIT: [Flutter](https://medium.com/flutter-community/flutter-on-desktop-a-real-competitor-to-electron-4f049ea6b061)

EDIT: webview along with others mentioned here: https://blog.stevensanderson.com/2019/11/01/exploring-lighter-alternatives-to-electron-for-hosting-a-blazor-desktop-app/

EDIT: https://cordova.apache.org/ and Ionic. Although these are for mobile apps only.

EDIT: WebView for Deno! https://deno.land/x/webview@0.4.7

EDIT: Electron "but easy" https://www.todesktop.com/

**Comment by danawoodman** [comment]
From my understanding Electrino is a proof of concept and not actually usable at least yet?

**Comment by taoeffect** [comment]
Unknown whether or not it's usable, but yeah it does look very early stage.

**Comment by danawoodman** [comment]
I was reading something a few weeks ago that said it is "demo-ware" so might be a bit down the road to use

**Comment by taoeffect** [comment]
I don't think it will be a big issue. It's just the native browser. The native browser has most / if not everything we need.

**Comment by vijayee** [comment]
I don't understand why we need a framework at all? We could use the base electron.

**Comment by taoeffect** [comment]
@vijayee Not sure what you mean by framework. Electrino is just a lighterweight alternative to bundling the site into a desktop app with Electron. Electron bundles the browser, Electrino doesn't.

**Comment by taoeffect** [comment]
Edited to add [Flutter](https://medium.com/flutter-community/flutter-on-desktop-a-real-competitor-to-electron-4f049ea6b061) to issue title.
